client.go

```diff
@@ -221,6 +231,12 @@ func (d *Dialer) Dial(urlStr string, requestHeader http.Header) (*Conn, *http.Re
 		}
 	}
 
+	if d.CompressionSupported {
+		req.Header.Set("Sec-Websocket-Extensions", "permessage-defalte; server_no_context_takeover; client_no_context_takeover")
+	} else {
+		delete(req.Header, "Sec-Websocket-Extensions")
```

Comment:
Follow pattern for Sec-Websocket-Protocol above. The dialer should return an error instead of silently deleting the the application's header.